gix-blame/src/types.rs

```diff
@@ -81,60 +101,51 @@ impl BlameRanges {
     /// If no merge is possible, add it as a new range.
-    fn merge_range(&mut self, new_range: RangeInclusive<u32>) {
-        // Check if this range can be merged with any existing range
-        for range in &mut self.ranges {
-            // Check if ranges overlap or are adjacent
-            if new_range.start() <= range.end() && range.start() <= new_range.end() {
-                *range = *range.start().min(new_range.start())..=*range.end().max(new_range.end());
-                return;
+    fn merge_range(&mut self, new_range: Range<u32>) -> Result<(), Error> {
+        match self {
+            Self::PartialFile(ref mut ranges) => {
+                // Check if this range can be merged with any existing range
+                for range in &mut *ranges {
+                    // Check if ranges overlap
+                    if new_range.start <= range.end && range.start <= new_range.end {
+                        *range = range.start.min(new_range.start)..range.end.max(new_range.end);
+                        return Ok(());
+                    }
+                    // Check if ranges are adjacent
+                    if new_range.start == range.end || range.start == new_range.end {
+                        *range = range.start.min(new_range.start)..range.end.max(new_range.end);
+                        return Ok(());
+                    }
+                }
+                // If no overlap or adjacency found, add it as a new range
+                ranges.push(new_range);
+                Ok(())
             }
+            _ => Err(Error::InvalidOneBasedLineRange),
         }
-        // If no overlap found, add it as a new range
-        self.ranges.push(new_range);
     }
 
-    /// Convert the 1-based inclusive ranges to 0-based exclusive ranges.
-    ///
-    /// This is used internally by the blame algorithm to convert from git's line number format
-    /// to the internal format used for processing.
-    ///
-    /// # Errors
-    ///
-    /// Returns `Error::InvalidLineRange` if:
-    /// - Any range starts at 0 (must be 1-based)
-    /// - Any range extends beyond the file's length
-    /// - Any range has the same start and end
-    pub fn to_zero_based_exclusive(&self, max_lines: u32) -> Result<Vec<Range<u32>>, Error> {
-        if self.ranges.is_empty() {
-            let range = 0..max_lines;
-            return Ok(vec![range]);
-        }
-
-        let mut result = Vec::with_capacity(self.ranges.len());
-        for range in &self.ranges {
-            if *range.start() == 0 {
-                return Err(Error::InvalidLineRange);
-            }
-            let start = range.start() - 1;
-            let end = *range.end();
-            if start >= max_lines || end > max_lines || start == end {
-                return Err(Error::InvalidLineRange);
+    /// Convert the ranges to a vector of `Range<u32>`.
+    pub fn to_ranges(&self, max_lines: u32) -> Vec<Range<u32>> {
```

Comment:
Just to be explicit, what do you think about calling this `to_one_based_exclusive_ranges`?